File: BOJ/13. 피보나치 수/1003_피보나치 함수.c
//https://www.acmicpc.net/problem/1003
#include <stdio.h> 

int main(void)
{
	int T, N,b=1,c;
	int a[50]= {0,1};
	scanf("%d", &T);
	for (int i = 0; i < T; i++)
	{
		
		scanf("%d", &N);
		for (int j = 2; j <= N; j++)
		{
			a[j] = a[j-1]+a[j-2];
		}
		if (N == 0)
		{
			b = 1;
			c = 0;
		}
		else
		{
			b = a[N-1];
			c = a[N];
		}
			
		
		printf("%d %d\n", b, c);
	}
	return 0;
}

File: BOJ/1. 단계별로 풀어보기/15. 동적 계획법/1003 피보나치 함수.py
#https://www.acmicpc.net/problem/
import sys
input = sys.stdin.readline

def main() :
    arr = [[1, 0], [0, 1]]
    for i in  range(2, 40) :
        zero = arr[i-1][0] + arr[i-2][0]
        one = arr[i-1][1] + arr[i-2][1]
        arr.append([zero, one])
        
    for _ in range(int(input())) :
        n = int(input())
        print(arr[n][0], arr[n][1])
        
main()
